tiledb/api/c_api/array_schema/CMakeLists.txt

```diff
@@ -36,6 +36,7 @@ commence(object_library capi_array_schema_stub)
   this_target_sources(${SOURCES})
   this_target_link_libraries(export)
   this_target_object_libraries(array_schema)
+  this_target_object_libraries(array)
```

Comment:
This is because `tiledb/sm/array_schema/array_schema.cc` calls upon `tiledb/sm/array/array_directory.cc`. So now the unit test needs to link with that.